### PR TITLE
RUMM-2922 Introduce `RUM.enable(with:in:)` API for 2.x

### PR DESCRIPTION
**Pull request by ncreated**
### What and why?

📦 This is second (after #1310) PR extracting the new set of public RUM APIs for `2.0.0`.

The new API of RUM will consist of 3 main interfaces:
```swift
// Configuration:
let config: RUMConfiguration

// Feature:
RUM.enable(with: config)

// Monitor API:
RUMMonitor.shared() -> RUMMonitorProtocol
```
This PR adds `RUM.enable(with:)`. Changes to `RUMConfiguration` will come in next PR.

### How?

Adding `RUM.enable(with:in:)` API removing obsolete `RUMMonitor.initialize(with:in:)`.

Except introducing new API:
* I added tests to RUM's request builder (`DatadogRUMTests.RequestBuilderTests`) as they were missing,
* Revamped `RUMInstrumentationTests` by also moving them from `DatadogTests` to `DatadogRUMTests`,
* (many) smaller updates to tests to use new initialization flow for RUM feature.

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests
- [x] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

File: DatadogRUM/Sources/Feature/DatadogRUMFeature.swift [hunk 1]
```diff
@@ -7,78 +7,128 @@
 import Foundation
 import DatadogInternal
 
+internal typealias RUMFeature = DatadogRUMFeature
+
+// TODO: RUMM-2922 Rename to `RUMFeature`
 internal final class DatadogRUMFeature: DatadogRemoteFeature {
```
Comment (ncreated) [inline DatadogRUM/Sources/Feature/DatadogRUMFeature.swift:13]: I'll resolve it in next PR - `RUMM-2922` is not yet over.

File: DatadogRUM/Sources/Instrumentation/RUMInstrumentation.swift
Comment (ncreated) [inline DatadogRUM/Sources/Instrumentation/RUMInstrumentation.swift:1]: No real changes in RUM instrumentation - just aligning things in this file for more coherence. Notable change is the removal of `enable()` + `deinitialize()` APIs, as now this component manages instrumentation between `init()` and `deinit()`.

File: DatadogRUM/Sources/RUMConfiguration.swift
Comment (ncreated) [inline DatadogRUM/Sources/RUMConfiguration.swift:1]: Real work in this file will happen in next PR. Here just doing partial step.

File: Tests/DatadogTests/Datadog/RUM/RUMFeatureTests.swift [hunk 2]
```diff
@@ -108,59 +105,6 @@ class RUMFeatureTests: XCTestCase {
         XCTAssertEqual(request.allHTTPHeaderFields?["DD-REQUEST-ID"]?.matches(regex: .uuidRegex), true)
     }
 
-    // MARK: - HTTP Message with included variant
-
-    func testItUsesExpectedVariantQueryParaemterInMessage() throws {
-        let server = ServerMock(delivery: .success(response: .mockResponseWith(statusCode: 200)))
-        let httpClient = HTTPClient(session: server.getInterceptedURLSession())
-
-        let randomVariant: String = .mockAny()
```
Comment (ncreated) [inline Tests/DatadogTests/Datadog/RUM/RUMFeatureTests.swift:117]: Moved to RUM's `RequestBuilderTests`. The only difference in this test and the next one is adding `variant` in URL query. Works much better if tested in pure unit-fashion.

**Review by maxep — CHANGES_REQUESTED** [review]
Nice work, it's taking shape!

I've left to remove the `RUMMonitorInternalProtocol` and the `monitorFactory`, but maybe I'm missing some needs for further PRs. LMK!

File: DatadogRUM/Sources/Feature/DatadogRUMFeature.swift [hunk 3]
```diff
@@ -7,78 +7,128 @@
...
     static let name = "rum"
 
     let requestBuilder: FeatureRequestBuilder
 
     let messageReceiver: FeatureMessageReceiver
 
-    let monitor: Monitor
+    let monitor: RUMMonitorProtocol & RUMMonitorInternalProtocol
```
[the remark this answers is not in the document]
Comment (ncreated) [reply DatadogRUM/Sources/Feature/DatadogRUMFeature.swift]: It made some sense when I started this task 🙃, but I agree it doesn't bring anything right now. Removed ✅, it's `monitor: Monitor` again.

File: DatadogRUM/Sources/Feature/DatadogRUMFeature.swift [hunk 4]
```diff
@@ -7,78 +7,128 @@
...
+            in: core,
+            configuration: configuration
+        )
+    }
 
+    private init(
+        with monitor: RUMMonitorProtocol & RUMMonitorInternalProtocol & RUMCommandSubscriber,
+        in core: DatadogCoreProtocol,
+        configuration: RUMConfiguration
+    ) throws {
```
[the remark this answers is not in the document]
Comment (ncreated) [reply DatadogRUM/Sources/Feature/DatadogRUMFeature.swift]: Agree 👍, let's keep it simple, removed ✅. Especially that soon, with [RUMM-3151](https://datadoghq.atlassian.net/browse/RUMM-3151) we won't need to patch `Monitor` dependencies anymore 🙌.

**Review by maxep — APPROVED** [review]
🚀

**Pull request merged**